packages/components/src/components/hds/form/label/index.hbs

```diff
@@ -3,7 +3,7 @@
   Copyright (c) HashiCorp, Inc.
   SPDX-License-Identifier: MPL-2.0
 }}
-<label class={{this.classNames}} for={{@controlId}} ...attributes>
+<label class={{this.classNames}} for={{@controlId}} id={{this.id}} ...attributes>
```

Comment:
if they assign an `id` to the Label (as for the HelperText, the CharacterCount, or the Error) they would need to wire the Label with the control themselves. the `Field` takes care of these links, but only if the user doesn't override the `id`'s of these subcomponents. that's how it works for all Form components. They can (and probably are) override the control `id` and the links would use that `id` instead of generating one.